tests/unit/cluster/replica-detach.tcl

```diff
@@ -0,0 +1,50 @@
+proc find_slots_for_node {node_idx node_id_to_find} {
+    set shards_cfg [R $node_idx CLUSTER SHARDS]
+    foreach shard_cfg $shards_cfg {
+        set nodes [dict get $shard_cfg nodes]
+        foreach node $nodes {
+            if {[dict get $node id] eq $node_id_to_find} {
+                return [dict get $shard_cfg slots]
+            }
+        }
+    }
+    return [list]
+}
+
+start_cluster 3 3 {tags {external:skip cluster}} {
+    test "Cluster should start ok" {
+        wait_for_cluster_state ok
+    }
+
+    test "CLUSTER REPLICATE NO ONE should turn node into empty primary" {
+        set replica_node 3
+        set replica_node_id [R $replica_node CLUSTER MYID]
+        # make sure node is replica
+        wait_for_condition 100 100 {
+            [string first "slave" [R $replica_node ROLE]] >= 0
+        } else {
+            puts "R $replica_node ROLE: [R $replica_node ROLE]"
+            fail "R $replica_node didn't assume replica role in time"
+        }
+
+        assert_equal "OK" [R $replica_node CLUSTER REPLICATE NO ONE]
+
+        # make sure node is turned into primary
+        wait_for_condition 100 100 {
+            [string first "master" [R $replica_node ROLE]] >= 0
+        } else {
+            puts "R $replica_node ROLE: [R $replica_node ROLE]"
+            fail "R $replica_node didn't assume primary role in time"
+        }
+
+        # checking that new primary has no slots across all nodes
```

Comment:
let's also check that the shard id is changed

Reply:
Done